llvm/include/llvm/SandboxIR/SandboxIR.h

```diff
@@ -1029,6 +1033,66 @@ class CallInst final : public CallBase {
 #endif
 };
 
+class InvokeInst final : public CallBase {
+  /// Use Context::createInvokeInst(). Don't call the
+  /// constructor directly.
+  InvokeInst(llvm::Instruction *I, Context &Ctx)
+      : CallBase(ClassID::Invoke, Opcode::Invoke, I, Ctx) {}
+  friend class Context; // For accessing the constructor in
+                        // create*()
+  Use getOperandUseInternal(unsigned OpIdx, bool Verify) const final {
+    return getOperandUseDefault(OpIdx, Verify);
+  }
+  SmallVector<llvm::Instruction *, 1> getLLVMInstrs() const final {
+    return {cast<llvm::Instruction>(Val)};
+  }
+
+public:
+  static InvokeInst *create(FunctionType *FTy, Value *Func,
+                            BasicBlock *IfNormal, BasicBlock *IfException,
+                            ArrayRef<Value *> Args, BBIterator WhereIt,
+                            BasicBlock *WhereBB, Context &Ctx,
+                            const Twine &NameStr = "");
+  static InvokeInst *create(FunctionType *FTy, Value *Func,
+                            BasicBlock *IfNormal, BasicBlock *IfException,
+                            ArrayRef<Value *> Args, Instruction *InsertBefore,
+                            Context &Ctx, const Twine &NameStr = "");
+
```

Comment:
Why is InvokeInstr missing a constructor with "InsertAtEnd" parameter?  CallInstr has it.

Reply:
So the issue is that in other instructions we had just two `create()` functions, one with `InsertBefore` and one with `InsertAtEnd`. But both of these can be replaced by one create() function that takes as arguments a `BasicBlock::Iterator  WhereIt` and a `BasicBlock *WhereBB`. Anyway, it's probably OK to either keep all three variants, or keep only one. I would need to revisit them with a refactoring patch. Any preference?

But anyway, I will add the missing `create()` function with the `InsertAtEnd` parameter.

Reply:
I don't have a preference but the LLVM IR has InsertPosition.  Should we just create one for SandboxIR too?

Reply:
Yes, we should introduce a similar class in SandboxIR.